apollo-federation/src/connectors/json_selection/methods/public/entries.rs

```diff
@@ -164,10 +165,9 @@ fn entries_shape(
         _ => Shape::error(
             format!("Method ->{} requires an object input", method_name.as_ref()),
             {
-                input_shape
-                    .locations
-                    .extend(method_name.shape_location(context.source_id()));
-                input_shape.locations
+                let mut locations = input_shape.locations().cloned().collect::<IndexSet<_>>();
+                locations.extend(method_name.shape_location(context.source_id()));
+                locations.into_iter()
             },
```

Comment:
```suggestion
            input_shape
                .locations()
                .cloned()
                .chain(method_name.shape_location(context.source_id())),
```

Saves 1 or 2 allocations. Unless you needed the deduplication from the IndexSet